src/Storage/Bucket.php

```diff
@@ -515,6 +529,167 @@ function (array $object) {
     }
 
     /**
+     * Create a Cloud PubSub notification.
+     *
+     * Example:
+     * ```
+     * // Assume the topic uses the same project ID as that configured on the
+     * // existing client.
+     * $notification = $bucket->createNotification('my-topic');
+     * ```
+     *
+     * ```
+     * // Use a fully qualified topic name.
+     * $notification = $bucket->createNotification('projects/my-project/topics/my-topic');
+     * ```
+     *
+     * ```
+     * // Provide a Topic object from the Cloud PubSub component.
+     * $topic = $pubSub->topic('my-topic');
+     * $notification = $bucket->createNotification($topic);
+     * ```
+     *
+     * ```
+     * // Supplying event types to trigger the notifications.
+     * $notification = $bucket->createNotification('my-topic', [
+     *     'event_types' => [
+     *         'OBJECT_DELETE',
+     *         'OBJECT_METADATA_UPDATE'
+     *     ]
+     * ]);
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @see https://cloud.google.com/storage/docs/pubsub-notifications Cloud PubSub Notifications
+     * @see https://cloud.google.com/storage/docs/json_api/v1/notifications/insert Notifications insert API documentation.
+     * @codingStandardsIgnoreEnd
+     *
+     * @param string|Topic $topic The topic used to publish notifications.
+     * @param array $options [optional] {
+     *     Configuration options.
+     *
+     *     @type array $custom_attributes An optional list of additional
```

Comment:
curious why these are snake case rather than our usual camelCase convention?

Reply:
The upstream docs use snake case, and I was wondering the same thing myself! I debated normalizing to camel case for the library and then converting to snake case during transport, but wasn't sure if it was worth the overhead. WDYT?